pkg/client/tests/remotecommand_test.go

```diff
@@ -188,116 +188,114 @@ func TestStream(t *testing.T) {
 	}
 
 	for _, testCase := range testCases {
-		for _, exec := range []bool{true, false} {
-			var name string
-			if exec {
-				name = testCase.TestName + " (exec)"
-			} else {
-				name = testCase.TestName + " (attach)"
-			}
-			var (
-				streamIn             io.Reader
-				streamOut, streamErr io.Writer
-			)
-			localOut := &bytes.Buffer{}
-			localErr := &bytes.Buffer{}
-
-			requestReceived := make(chan struct{})
-			server := httptest.NewServer(fakeServer(t, requestReceived, name, exec, testCase.Stdin, testCase.Stdout, testCase.Stderr, testCase.Error, testCase.Tty, testCase.MessageCount, testCase.ServerProtocols))
-
-			url, _ := url.ParseRequestURI(server.URL)
-			config := restclient.ClientContentConfig{
-				GroupVersion: schema.GroupVersion{Group: "x"},
-				Negotiator:   runtime.NewClientNegotiator(legacyscheme.Codecs.WithoutConversion(), schema.GroupVersion{Group: "x"}),
-			}
-			c, err := restclient.NewRESTClient(url, "", config, nil, nil)
-			if err != nil {
-				t.Fatalf("failed to create a client: %v", err)
-			}
-			req := c.Post().Resource("testing")
+		t.Run(testCase.TestName, func(t *testing.T) {
+			for _, exec := range []bool{true, false} {
+				var name string
+				if exec {
+					name = testCase.TestName + " (exec)"
+				} else {
+					name = testCase.TestName + " (attach)"
+				}
```

Comment:
are  these independent test cases too?

Reply:
I mean, this for loop should go out of the t.Run too, no?

Reply:
Actually, it's independent, updated with the latest commit. Also update the `TestForwardPorts` function, which defer to close the client conn in each loop.